### PR TITLE
init: honeypot field

### PR DESCRIPTION
**Pull request by bobmitch**

### CODE REVIEW

**Comment by workaj** [comment]
no

File: core/fields/Field_Honeypot.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		// tabindex is required as -1 for accessibility reasons, so might tip off some bots, but can't hurt screen readers etc
+		// set value to be ' ' (space) - allows us to use 'required' client-side, but whitespace might be more enticing to replace 
+		// set display/style attributes via js as another layer of obfuscation
+		?>
+		<input required placeholder="Important information" type='text' tabindex="-1" autocomplete="nothingtoseehere" id='<?php echo $this->id;?>' <?php echo $this->get_rendered_name();?> value=' '/>
+		<script>
+			let hp = document.getElementById('<?php echo $this->id;?>') ?? null;
+			if (hp) {
+				hp.style.position = 'absolute'; // take up no space in document flow
+				hp.style.clipPath = 'circle(0)'; // sneaky invisible method that might hide from most bots
```
Comment (workaj) [inline core/fields/Field_Honeypot.php:26]: should put the comments in `<?php //comment ?>` so they dont show up on the front end

File: core/fields/Field_Honeypot.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+		$this->save=false;
+	}
+
+	public function display() {
+		// autocomplete attribute set to nonsense which is calculated to be same as 'off' without being explicit
+		// tabindex is required as -1 for accessibility reasons, so might tip off some bots, but can't hurt screen readers etc
+		// set value to be ' ' (space) - allows us to use 'required' client-side, but whitespace might be more enticing to replace 
+		// set display/style attributes via js as another layer of obfuscation
+		?>
+		<input required placeholder="Important information" type='text' tabindex="-1" autocomplete="nothingtoseehere" id='<?php echo $this->id;?>' <?php echo $this->get_rendered_name();?> value=' '/>
```
Comment (workaj) [inline core/fields/Field_Honeypot.php:21]: should set autocomplete to name or something to make it seem more legit

**Review by workaj — COMMENTED** [review]
changes

File: core/fields/Field_Honeypot.php (hunk 2 above)
Comment (workaj) [reply core/fields/Field_Honeypot.php:21]: or make json customizable

Comment (bobmitch) [reply core/fields/Field_Honeypot.php]: now user configurable

**Review by workaj — APPROVED** [review]
seems legit

**Pull request merged**